### ISSUE TITLE
Rework API to work with System.Web.Abstractions

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Benknightdark/elmah#87]
```
Lots of HttpContext everywhere, but no HttpContextBase.  Refactor to use 
the HttpContextBase so we can substitute the HttpContextWrapper.



```

Original issue reported on code.google.com by `yesthatm...@gmail.com` on 15 Apr 2009 at 7:01

**Comment by GoogleCodeExporter** [comment]
```
Definitely, as soon as we can drop that .NET 1.x baggage after ELMAH 1.0 goes 
gold.
```

Original comment by `azizatif` on 15 Apr 2009 at 8:56
- Changed state: **Accepted**
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
Original comment by `azizatif` on 27 May 2009 at 11:08
- Added labels: **Milestone-Release2.0**

**Comment by GoogleCodeExporter** [comment]
Original comment by `azizatif` on 25 Dec 2011 at 1:38
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 72aab7535e19.
```

Original comment by `azizatif` on 25 Dec 2011 at 4:51
- Changed state: **Fixed**
